ffs/minerselector/sr2/sr2.go

```diff
@@ -0,0 +1,149 @@
+package sr2
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"math/rand"
+	"net/http"
+	"time"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/lotus/api/apistruct"
+	"github.com/filecoin-project/lotus/chain/types"
+	logger "github.com/ipfs/go-log/v2"
+	"github.com/textileio/powergate/ffs"
+	askindex "github.com/textileio/powergate/index/ask/runner"
+	"github.com/textileio/powergate/lotus"
+)
+
+var (
+	log = logger.Logger("sr2-miner-selector")
+)
+
+// MinerSelector chooses miner under SR2 strategy.
+type MinerSelector struct {
+	url string
+	ai  *askindex.Runner
+	cb  lotus.ClientBuilder
+}
+
+var _ ffs.MinerSelector = (*MinerSelector)(nil)
+
+type minersBuckets struct {
+	Buckets []bucket
+}
+
+type bucket struct {
+	Amount         int
+	MinerAddresses []string
+}
+
+// New returns a new SR2 miner selector.
+func New(url string, ai *askindex.Runner, cb lotus.ClientBuilder) (*MinerSelector, error) {
+	ms := &MinerSelector{url: url, ai: ai, cb: cb}
+
+	_, err := ms.getMiners()
+	if err != nil {
+		return nil, fmt.Errorf("verifying sr2 url content: %s", err)
+	}
+
+	return ms, nil
+}
+
+// GetMiners returns miners from SR2.
+func (ms *MinerSelector) GetMiners(n int, f ffs.MinerSelectorFilter) ([]ffs.MinerProposal, error) {
```

Comment:
Important fact. This MinerSelector will ignore `n` and `f`. 
The replication factor will be the sum of `Amount` of buckets, and not filtering is done.

So this is a very aggressive miner selector kind of forcing what was decided in SR2. No custom _StorageConfig_ will change this fact, unless the user runs with `--ffsminerselector=reputation`.

Reply:
Yes, is ignored. Now will be controled in this remote file now that `--ffsminerselector=sr2` default miner selector.
If someone wants to come back to not being ignored, they could specify `--ffsminerselector=reputation` and things will work as before.